nekoyume/Assets/_Scripts/State/LocalLayerActions.cs

```diff
@@ -0,0 +1,137 @@
+
+                return;
+            }
+
+            info = GetNewInfo(payCostAction, createdBlockIndex, isRendered);
+            _infos.Add(gameActionId, info);
+        }
+
+        /// <summary>
+        /// Set rendered or unrendered flag. 
+        /// </summary>
+        /// <param name="gameActionId">`GameAction.Id`</param>
+        /// <param name="isRendered">Set `True` when rendered. Set `False` when unrendered.</param>
+        public static void SetRendered(Guid gameActionId, bool isRendered)
+        {
+            if (!TryGetRegisteredInfo(gameActionId, out var info))
+            {
+                Debug.LogError($"[{nameof(LocalLayerActions)}] There is no registered. {gameActionId.ToString()}");
+
+                return;
+            }
+
+            info.isRendered = isRendered;
+        }
+
+        /// <summary>
+        /// Pay the cost of registered `GameAction`s which is not rendered.
+        /// </summary>
+        /// <param name="states">Source `States`</param>
+        /// <param name="tableSheets"></param>
+        public static void PayCost(States states, TableSheets tableSheets)
+        {
+            foreach (var info in _infos.Values.Where(e => !e.isRendered))
+            {
+                info.payCostAction(states, tableSheets);
+            }
+        }
+
+        /// <summary>
+        /// Unregister the registered `GameAction` information which is created before `blockIndex`.
+        /// </summary>
+        /// <param name="blockIndex"></param>
+        public static void UnregisterCreatedBefore(long blockIndex)
+        {
+            foreach (var gameActionId in _infos
+                .Where(pair => pair.Value.createdBlockIndex < blockIndex)
+                .Select(pair => pair.Key)
+                .ToArray())
+            {
+                var info = _infos[gameActionId];
+                _infos.Remove(gameActionId);
+                _reusableInfoPool.Push(info);
+            }
+        }
+
+        public static void UnregisterAll()
+        {
+            var infos = _infos.Values.ToArray();
+            _infos.Clear();
+            foreach (var info in infos)
+            {
+                _reusableInfoPool.Push(info);
+            }
+        }
+
+        private static bool TryGetRegisteredInfo(Guid gameActionId, out Info info)
+        {
+            if (_infos.ContainsKey(gameActionId))
+            {
+                info = _infos[gameActionId];
+                return true;
+            }
+
+            info = null;
+            return false;
+        }
+
+        private static Info GetNewInfo(Action<States, TableSheets> payCostAction, long createdBlockIndex,
+            bool isRendered = false)
+        {
+            // FIXME
+            var info = _reusableInfoPool.Count == 0
+                ? new Info
+                {
+                    payCostAction = payCostAction,
+                    createdBlockIndex = createdBlockIndex,
+                    isRendered = isRendered,
```

Comment:
I think fields are not need to be assigned in constructor because they're assigned below.